rollup-interface/src/state_machine/zk/mod.rs

```diff
@@ -43,10 +43,40 @@ pub trait Zkvm {
 
     /// Same as [`verify`], except that instead of returning the output as a serialized array,
     /// it returns a state transition structure.
-    fn verify_and_extract_output<C: ValidityCondition, Add: AddressTrait>(
+    /// TODO: specify a deserializer for the output
+    fn verify_and_extract_output<
+        C: ValidityCondition,
+        Add: AddressTrait + BorshDeserialize + BorshSerialize,
+    >(
         serialized_proof: &[u8],
         code_commitment: &Self::CodeCommitment,
-    ) -> Result<StateTransition<C, Add>, Self::Error>;
+    ) -> Result<StateTransition<C, Add>, Self::Error> {
+        let mut output = Self::verify(serialized_proof, code_commitment)?;
+        Ok(BorshDeserialize::deserialize_reader(&mut output)?)
+    }
+}
+
+/// A wrapper around a code commitment which implements borsh serialization
+#[derive(Clone, Debug)]
+pub struct StoredCodeCommitment<Vm: Zkvm> {
+    /// The inner field of the wrapper that contains the code commitment.
+    pub commitment: Vm::CodeCommitment,
+}
```

Comment:
Out-of-scope: we can get rid of this wrapper now that the JMT supports arbitrary serialization

Reply:
Let's add a small issue for that

Reply:
Remove StoredCodeCommitment wrapper #676